src/utilities/include/utilities/Interval.h

```diff
@@ -43,6 +47,124 @@ class Interval
 	{
 	}
 
+	/// Check if the first interval is smaller than the second.
+	/** Interval I1 is smaller than Interval I2 if
+	 * 1. the lower bound of I1 is smaller than the lower bound of I2,
+	 * 2. they have the same lower bound and the upper bound of I1 is smaller than the upper bound
+	 * of I2.
+	 * @param first The first interval to compare
+	 * @param second The second interval to compare
+	 * @return true if first is smaller than second
+	 */
+	friend bool
+	operator<(const Interval &first, const Interval &second)
+	{
+		auto bound_is_smaller = [](const N & first_value,
+		                           BoundType first_type,
+		                           const N & second_value,
+		                           BoundType second_type) -> bool {
+			if (first_type == BoundType::INFTY || second_type == BoundType::INFTY) {
+				// Bound values do not matter as at least one bound is infty.
+				// For bound types: WEAK < STRICT < INFTY
```

Comment:
Is Infty as a lower bound not always smaller than anything bounded?

Reply:
Fixed!